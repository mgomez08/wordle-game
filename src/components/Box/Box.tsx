import { BoxStatus } from "../../types/types";
import styles from "./box.module.scss";
import classNames from "classnames/bind";

const classes = classNames.bind(styles);

interface BoxProps {
  value: string;
  status: BoxStatus;
}

export const Box = ({ value, status }: BoxProps) => {
  const boxStatus = classes({
    absent: status === "absent",
    correct: status === "correct",
    edit: status === "edit",
    empty: status === "empty",
    present: status === "present",
  });
  return <div className={boxStatus}>{value}</div>;
};
