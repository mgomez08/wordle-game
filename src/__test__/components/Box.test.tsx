import { mount } from "enzyme";
import { Box } from "../../components/Box/Box";

describe("Tests for <Box />", () => {
  const box = mount(<Box value="H" status="correct" />);
  test("Render Box component", () => {
    expect(box.length).toEqual(1);
  });
  test("Box component should have correct class", () => {
    expect(box.find(".correct").length).toEqual(1);
  });
});
