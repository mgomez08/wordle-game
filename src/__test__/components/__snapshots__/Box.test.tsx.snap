// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Box /> Box component with Snapshot 1`] = `ReactWrapper {}`;
